packages/plugin/vite/src/Config.ts

```diff
@@ -0,0 +1,34 @@
+import type { LibraryOptions } from 'vite';
+
+export interface VitePluginBuildConfig {
+  /**
+   * Alias of `build.lib.entry` in `config`.
+   */
+  entry?: LibraryOptions['entry'];
+  /**
+   * Vite config file path.
+   */
+  config?: string;
+}
+
+export interface VitePluginRendererConfig {
+  /**
+   * Human friendly name of your entry point
+   */
+  name: string;
+  /**
+   * Vite config file path.
+   */
+  config: string;
+}
+
+export interface VitePluginConfig {
+  /**
+   * Build anything such as Main process, Preload scripts and Worker process, etc.
+   */
+  build: VitePluginBuildConfig[];
+  /**
+   * Renderer process, the first item will be used as the main entry.
```

Comment:
Is there a functional difference for the "main entry"? I saw in the `getRendererConfig()` function that we just iterate over the entire `config.renderer` array and assign a port to each.